pkg/controller/useraccount/useraccount_controller.go

```diff
@@ -608,3 +644,59 @@ func newNSTemplateSet(userAcc *toolchainv1alpha1.UserAccount) *toolchainv1alpha1
 func ToIdentityName(userID string, identityProvider IdentityProvider) string {
 	return fmt.Sprintf("%s:%s", identityProvider.GetIdP(), userID)
 }
+
+func (r *ReconcileUserAccount) lookupAndDeleteCheUser(userAcc *toolchainv1alpha1.UserAccount) error {
+
+	// If the admin username is not set then Che user deletion is not configured, just return
+	if r.config.GetCheAdminUsername() == "" {
+		log.Info("Che user deletion is not configured, configure the Che admin credentials to enable it")
+		return nil
+	}
+
+	var userExists bool
+
+	userExists, err := r.cheClient.UserExists(userAcc.Name)
+	if err != nil {
+		return err
+	}
+
+	// If the user doesn't exist and there's no deletion in progress then there's nothing left to do.
+	// We need to check whether the deletion is in progress because in some cases the user could be reported as non-existent while the deletion is still in progress.
+	if !userExists && !condition.HasConditionReason(userAcc.Status.Conditions, toolchainv1alpha1.UserAccountCheCleanup, toolchainv1alpha1.UserAccountDeletingCheDataReason) {
+		return nil
+	}
+
+	var cheUserID string
+
+	// Get the user ID from the status condition if it's there
+	if condition.HasConditionReason(userAcc.Status.Conditions, toolchainv1alpha1.UserAccountCheCleanup, toolchainv1alpha1.UserAccountDeletingCheDataReason) {
+		cond, found := condition.FindConditionByType(userAcc.Status.Conditions, toolchainv1alpha1.UserAccountCheCleanup)
+		if found {
+			cheUserID = cond.Message
+		}
+	}
+
+	// If the user ID is not in the status then look it up and set it in the status
+	if cheUserID == "" {
+		cheUserID, err = r.cheClient.GetUserIDByUsername(userAcc.Name)
+		if err != nil {
+			return err
+		}
+
+		// Set the Che user deletion status (store the Che user ID in the status for subsequent retries if needed)
+		// This is required because the deletion API will return an error until the user is successfully removed.
+		if !condition.HasConditionReason(userAcc.Status.Conditions, toolchainv1alpha1.UserAccountCheCleanup, toolchainv1alpha1.UserAccountDeletingCheDataReason) {
+			if err := r.setStatusCheUserDeletionInProgress(userAcc, cheUserID); err != nil {
+				return err
+			}
+		}
+	}
+
+	// Delete the Che user. It is common for this call to return an error multiple times before succeeding
+	if err := r.cheClient.DeleteUser(cheUserID); err != nil {
+		return err
```

Comment:
It's worth wrapping this error with an explanation that that error is kinda normal, so we don't get scared if see it in the logs and in the ua status.

Reply:
Good point! Done

Reply:
yes, something like `Keep calm` or `Stay cool` 😂